protostuff-runtime/src/main/java/io/protostuff/runtime/RuntimeCollectionFieldFactory.java

```diff
@@ -506,4 +505,20 @@ public FieldType getFieldType()
         }
     };
 
+    @SuppressWarnings("unchecked")
+    private static<T extends Collection<?>, E extends Enum<E>> void merge(Accessor accessor,
+                                                                          Object message,
+                                                                          Input input,
+                                                                          Schema<T> schema,
+                                                                          MessageFactory messageFactory,
+                                                                          Class<E>... enumType) throws IOException
+    {
+        T collection = accessor.get(message);
+        if (enumType.length == 0)
```

Comment:
This change will introduce a significant performance penalty, as for every element of collection you call `removeUnmodifiableWrapper`. Probably there is better solution, like catch `UnsupportedOperationException` when trying to add element to a collection field and creating new collection instance in that case.